src/Microsoft.ML.Data/Scorers/FeatureContributionCalculationTransform.cs

```diff
@@ -420,77 +420,5 @@ public Row GetRow(Row input, Func<int, bool> active)
                 yield return RoleMappedSchema.ColumnRole.Feature.Bind(InputRoleMappedSchema.Feature.Name);
             }
         }
-
-        private sealed class FeatureContributionSchema : ISchema
-        {
-            private readonly Schema _parentSchema;
-            private readonly IExceptionContext _ectx;
-            private readonly string[] _names;
-            private readonly ColumnType[] _types;
-            private readonly Dictionary<string, int> _columnNameMap;
-            private readonly int _featureCol;
-            private readonly int _featureVectorSize;
-            private readonly bool _hasSlotNames;
-
-            public int ColumnCount => _types.Length;
-
-            public FeatureContributionSchema(IExceptionContext ectx, string columnName, ColumnType columnType, Schema parentSchema, int featureCol)
-            {
-                Contracts.CheckValueOrNull(ectx);
-                Contracts.CheckValue(parentSchema, nameof(parentSchema));
-                _ectx = ectx;
-                _ectx.CheckNonEmpty(columnName, nameof(columnName));
-                _parentSchema = parentSchema;
-                _featureCol = featureCol;
-                _featureVectorSize = _parentSchema.GetColumnType(_featureCol).VectorSize;
-                _hasSlotNames = _parentSchema.HasSlotNames(_featureCol, _featureVectorSize);
-
-                _names = new string[] { columnName };
-                _types = new ColumnType[] { columnType };
-                _columnNameMap = new Dictionary<string, int>() { { columnName, 0 } };
-            }
-
-            public bool TryGetColumnIndex(string name, out int col)
-            {
-                return _columnNameMap.TryGetValue(name, out col);
-            }
-
-            public string GetColumnName(int col)
-            {
-                _ectx.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-                return _names[col];
-            }
-
-            public ColumnType GetColumnType(int col)
-            {
-                _ectx.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-                return _types[col];
-            }
-
-            public IEnumerable<KeyValuePair<string, ColumnType>> GetMetadataTypes(int col)
-            {
-                _ectx.CheckParam(col == 0, nameof(col));
-                if (_hasSlotNames)
-                    yield return MetadataUtils.GetSlotNamesPair(_featureVectorSize);
-            }
```

Comment:
that stuff would propagate slotnames for new added column from feature column.
I don't think you do that in new output schema.

Reply:
[This PR](https://github.com/dotnet/machinelearning/pull/1847/files) is using the same strategy, so there is no reason to maintain that behavior.